bowler/tool.py

```diff
@@ -281,14 +283,19 @@ def process_hunks(self, filename: Filename, hunks: List[Hunk]) -> None:
                         raise ValueError("unknown response")
 
             if result == "y" or self.write:
-                patch = ("\n".join(hunk) + "\n").encode("utf-8")
-                args = ["patch", "-u", filename]
-                self.log_debug(f"running {args}")
-                try:
-                    sh.patch("-u", filename, _in=patch)  # type: ignore
-                except sh.ErrorReturnCode:
-                    log.exception("failed to apply patch hunk")
-                    return  # TODO: better response?
+                need_patch = True
+                accepted_hunks += "\n".join(hunk[2:]) + "\n"
+
+        if need_patch:
```

Comment:
nit: Since this is the end of the function, you can do "`if not need_patch: return`" instead to save an indentation level and make it clearer to the reader that this situation is already handled.

Better yet, you could skip the need for `need_patch` altogether if you start with an empty `accepted_hunks` string and only add the header if `accepted_hunks` is non-empty after the for-loop.